utoipa-gen/src/schema/into_params.rs

```diff
@@ -315,30 +315,30 @@ struct ParamType<'a>(&'a ComponentPart<'a>);
 
 impl ToTokens for ParamType<'_> {
     fn to_tokens(&self, tokens: &mut proc_macro2::TokenStream) {
-        let ty = self.0;
-        match &ty.generic_type {
+        let part: &ComponentPart = self.0;
+        match &part.generic_type {
             Some(GenericType::Vec) => {
-                let param_type = ParamType(ty.child.as_ref().unwrap().as_ref());
+                let param_type = ParamType(part.child.as_ref().unwrap().as_ref());
 
                 tokens.extend(quote! { #param_type.to_array_builder() });
             }
-            None => match ty.value_type {
+            None => match part.value_type {
                 ValueType::Primitive => {
-                    let component_type = ComponentType(ty.ident);
+                    let component_type = ComponentType(&*part.path);
 
                     tokens.extend(quote! {
                         utoipa::openapi::PropertyBuilder::new().component_type(#component_type)
                     });
 
-                    let format = ComponentFormat(ty.ident);
+                    let format = ComponentFormat(&*part.path);
                     if format.is_known_format() {
                         tokens.extend(quote! {
                             .format(Some(#format))
                         })
                     }
                 }
                 ValueType::Object => {
-                    let name = ty.ident.to_string();
+                    let name = format_path_ref(&part.path.to_token_stream().to_string());
```

Comment:
Rust guidelines usually tells to call free standing functions with the path. This gives user more context where the function is located and user will not mix it with local functions. :)
```suggestion
                    let name = component::format_path_ref(&part.path.to_token_stream().to_string());
```